streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -144,6 +144,13 @@ class StreamingContext private[streaming] (
     }
   }
 
+  if (sc.conf.contains("spark.cores.max")) {
+    val totalCores = sc.conf.getInt("spark.cores.max", 1)
```

Comment:
AFAIK, "spark.cores.max" is not set by default, with your change it will always return 1 if not set and throws exception as you wrote.

Besides if there're several receivers in streaming application, even > 1 is not sufficient. So simply checking "spark.cores.max" is not a feasible way to address the issue, and there may not be a good way to handle this problem properly, so I'd suggest to leave the current code as it is.